src/Telegram.Bot/TelegramBotClient.cs

```diff
@@ -406,22 +392,21 @@ public void StopReceiving()
         /// </remarks>
         /// <returns>An Array of <see cref="Update"/> is returned.</returns>
         /// <see href="https://core.telegram.org/bots/api#getupdates"/>
-        public Task<Update[]> GetUpdatesAsync(int offset = 0, int limit = 100, int timeout = 0,
-            UpdateType[] allowedUpdates = null,
-            CancellationToken cancellationToken = default)
-        {
-            var parameters = new Dictionary<string, object>
+        public Task<Update[]> GetUpdatesAsync(
+            int offset = default,
+            int limit = default,
+            int timeout = default,
+            IEnumerable<UpdateType> allowedUpdates = default,
+            CancellationToken cancellationToken = default
+        ) =>
+            MakeRequestAsync(new GetUpdatesRequest
             {
-                {"offset", offset},
-                {"limit", limit},
-                {"timeout", timeout}
-            };
-
-            if (allowedUpdates != null && !allowedUpdates.Contains(UpdateType.All))
-                parameters.Add("allowed_updates", allowedUpdates);
-
-            return SendWebRequestAsync<Update[]>("getUpdates", parameters, cancellationToken);
-        }
+                Offset = offset,
+                Limit = limit,
+                Timeout = timeout,
+                AllowedUpdates = allowedUpdates
+            }, cancellationToken)
+            .ContinueWith(t => t.Result.ToArray());
```

Comment:
Do you remember why we decided to use `async`/`await` instead of `ContinuteWith`? `ContinueWith` throws `AggregateException` with `ApiResponseException` inside `InnerExceptions` collection so we have to unwrap it ourselves and rethrow. With `async`/`await` we can be sure that the compiler generates the necessary code.

Reply:
That's right. In fact, we should just return the `List<T>` itself. I'll fix it.